README.md

```diff
@@ -1,94 +1,132 @@
-# Go Test WAF
+# GoTestWAF
 
-GoTestWAF is a tool for API and OWASP attack simulation, that supports a wide range of API protocols including
+GoTestWAF is a tool for API and OWASP attack simulation that supports a wide range of API protocols including
 REST, GraphQL, gRPC, WebSockets, SOAP, XMLRPC, and others.
 
-It was designed to evaluate web application security solutions, such as API security proxies, Web Application Firewalls, IPS, API gateways, and others.
+It was designed to evaluate web application security solutions, such as API security proxies, Web Application Firewalls,
+IPS, API gateways, and others.
 
-# How it works
+## How it works
 
-It is a 3-steps requests generation process that multiply amount of payloads to encoders and placeholders.
-Let's say you defined 2 payloads, 3 encoders (Base64, JSON, and URLencode) and 1 placeholder (HTTP GET variable).
-In this case, the tool will send 2x3x1 = 6 requests in a testcase.
+GoTestWAF generates malicious requests using encoded payloads placed in different parts of HTTP requests: its body, headers,
+URL parameters, etc. Generated requests are sent to the application security solution URL specified during GoTestWAF launch.
+The results of the security solution evaluation are recorded in the report file created on your machine.
 
-## Payload
+Default conditions for request generation are defined in the `testcases` folder in the YAML files of the following format:
 
-The payload string you wanna send. Like ```<script>alert(111)</script>``` or something more sophisticated.
-There is no macroses like so far, but it's in our TODO list. 
-Since it's a YAML string, use binary encoding if you wanna to https://yaml.org/type/binary.html
+```
+payload:
+  - '"union select -7431.1, name, @aaa from u_base--w-'
+  - "'or 123.22=123.22"
+  - "' waitfor delay '00:00:10'--"
+  - "')) or pg_sleep(5)--"
+encoder:
+  - Base64Flat
+  - URL
+placeholder:
+  - UrlPath
+  - UrlParam
+  - JSUnicode
+  - Header
+```
 
-## Encoder
+* `payload` is a malicious attack sample (e.g XSS string like ```<script>alert(111)</script>``` or something more sophisticated).
```

Comment:
XSS vector

Reply:
XSS string is ok too (as a synonym or a word form)